qutip/qip/circuit_latex.py

```diff
@@ -42,31 +47,121 @@
 """
 
 
-def _latex_compile(code, filename="qcirc", format="png"):
+def _find_system_command(names):
+    for name in names:
+        if shutil.which(name) is not None:
+            return name
+    return None
+
+
+_pdflatex = _find_system_command(['pdflatex'])
+_pdfcrop = _find_system_command(['pdfcrop'])
+
+
+def _force_remove(*filenames):
+    """rm -f`: try to remove a file, ignoring errors if it doesn't exist."""
+    for filename in filenames:
+        try:
+            os.remove(filename)
+        except FileNotFoundError:
+            pass
+
+
+_run_command = functools.partial(subprocess.run, check=True,
+                                 stdout=subprocess.DEVNULL,
+                                 stderr=subprocess.DEVNULL)
+_run_command.__doc__ = \
+    """
+    Run a command with stdout and stderr explicitly thrown away, raising
+    `subprocess.CalledProcessError` if the command returned a non-zero exit
+    code.
+    """
+
+if not _pdfcrop:
+    def _crop_pdf(_):
+        warnings.warn("Could not locate system 'pdfcrop':"
+                      " image output may have additional margins.")
+else:
+    def _crop_pdf(filename):
+        temporary = ".tmp." + filename
+        _run_command((_pdfcrop, filename, temporary))
+        os.rename(temporary, filename)
```

Comment:
I got an error there saying `[WinError 183] Cannot create a file when that file already exists: '.tmp.qcirc.pdf' -> 'qcirc.pdf'`. I need to add `_force_remove(filename)` before renaming to delete `qcirc.pdf`.

Reply:
Thanks - I've not got a Windows box, and I forgot that it doesn't have the "move is overwrite" semantics that unix derivatives do.